app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt

```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
             SyncResult.fromValue(true)
         }
     }
+    override fun add(login: ServerPassword): SyncResult<String> {
+        return list().then {
+            if (login.id == "") {
+                SyncResult.fromException<String>(InvalidRecordException("record id missing"))
+            } else if (null != it.find { it.id == login.id }) {
+                SyncResult.fromException<String>(IdCollisionException("record already exists"))
+            }
+
+            all.add(login)
```

Comment:
What is `all`?

Reply:
all is the MutableList of credential instances.

Reply:
👍